errors.go

```diff
@@ -133,3 +133,56 @@ func ErrorIsAll(expected ...error) ErrorCheck {
 		}
 	}
 }
+
+// Must can be used on a (value, error) pair to either get the value or
+// immediately fail the test if the error is non-nil. The T parameter is
+// curried, rather than passed as a third argument, so that (value, error)
+// function return values can be passed to Must directly, without assigning them
+// to intermediate variables.
+//
+// See also Must0, Must2, and Must3 for working with functions of other coarity.
+//
+//	bytes := expect.Must(t)(io.ReadAll(reader))
+func Must[V any](value V, err error) func(T) V {
+	return func(t T) V {
+		t.Helper()
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+		return value
+	}
+}
+
+// Must0 is similar to Must but for functions returning just an error, without a
+// value.
+func Must0(err error) func(T) {
```

Comment:
This function wouldn't have to be curried since a unary function can be passed directly even when there are other parameters. However, opted to follow the pattern of the other functions for consistency.

Reply:
Definitely took me a minute to really understand why we're returning an unary function of T. But now I get it. I kinda long for a type-safe arg spread operator. this is a reasonably un-awkward solution.